core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -1092,37 +1093,38 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
       jobFormat.checkOutputSpecs(job)
     }
 
+    // Instantiate writer
+    val committer = FileCommitProtocol.instantiate(
+      className = classOf[HadoopMapReduceCommitProtocol].getName,
+      jobId = stageId.toString,
```

Comment:
In fact I'm not sure what value should be assigned to `jobId` here, should it be `jobTrackerId` combined with `jobId`, or any value else? I failed to find some code to follow on this topic.